kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -203,34 +203,30 @@ showLeafs = do
     showPair (ns, xs) = show ns <> ": " <> show xs
 
 showPrecBranch
-    :: MetaOrObject level
-    => Maybe Int
+    :: Maybe Int
     -> StateT (ReplState level) Simplifier ()
 showPrecBranch mnode = do
     Strategy.ExecutionGraph { graph } <- Lens.use lensGraph
     node <- Lens.use lensNode
     let node' = maybe node id mnode
     if node' `elem` Graph.nodes graph
-        then do
-            putStrLn' $ show (findBranch graph [node'])
-        else putStrLn' "Invalid node!"
+       then putStrLn' . show $ loop (loopCond graph) node'
+       else putStrLn' "Invalid node!"
   where
-    findBranch gph xs
-      | xs == [] = []
-      | (Graph.outdeg gph) (head xs) <= 1 = (findBranch gph) ([head xs] >>= (Graph.pre gph))
-      | otherwise = xs
+    loopCond gph n
+      | (Graph.outdeg gph n) <= 1 && (not . null) (Graph.pre gph n)
+          = Left $ head (Graph.pre gph n)
+      | otherwise = Right [n]
```

Comment:
Does it still make sense to have this as an array? Can't it be just single value?